### PR TITLE
Add 1D cooling Thermo-Mechanical tutorial to documentation

### PR DESCRIPTION
**Pull request by npillardou**
This PR adds a new validation example under `advancedExamples/validationStudies/thermoPoromechanics/`
  that isolates a single mechanism: cooling a rock that is not free to contract generates
  tensile stress, and the constitutive model decides whether that stress leads to failure.

  The example reuses the `1DCooling` input files already in `develop` and contrasts two runs
  that differ **only** by the solid model:

  - `ThermoElastic_1DCooling_fim_smoke.xml` — `ElasticIsotropic`: the induced stress grows
    linearly with cooling, without bound.
  - `ThermoDruckerPrager_1DCooling_fim_smoke.xml` — `DruckerPrager`: the stress saturates on
    the yield surface and the rock deforms plastically.

  ### Documentation infrastructure

  Follows the convention introduced in #3884: the `.hdf5` outputs are **not** versioned.

  - `postprocess1DCooling.py` — run locally, reads the TimeHistory outputs and writes a single
    20 kB CSV.
  - `plot1DCooling.py` — run by Sphinx at build time, reads that CSV only. Depends on `numpy`
    and `matplotlib` alone, no `h5py` and no `mpmath`.

  ### Files

  | file | change |
  |---|---|
  | `1DCooling/Example.rst` | new tutorial page |
  | `1DCooling/AnalyticalSol.py` | reference analytical solution |
  | `1DCooling/postprocess1DCooling.py` | new, local post-processing |
  | `1DCooling/plot1DCooling.py` | new, build-time figure |
  | `1DCooling/cooling1D.csv` | new, plotted data (20 kB) |
  | `thermoPoromechanics/Index.rst` | toctree entry |
  | 3 × `inputFiles/thermoPoromechanics/*1DCooling*.xml` | SPHINX comment markers only |

  ### Notes for reviewers

  - The three input files are touched **only** to add `<!-- SPHINX_... -->` comment markers, so
    the tutorial can `literalinclude` the constitutive and boundary-condition blocks directly
    from the XML rather than duplicating them. No physics, mesh, solver or event is changed.
  - No new integrated test: the three `1DCooling` decks are already declared in
    `thermoPoromechanics.ats` on `develop`.
  - To regenerate the CSV after a change to the cases, run both decks in their own directories
    and call `postprocess1DCooling.py -e <elastic_dir> -d <druckerPrager_dir>`.

### CODE REVIEW

**Review by frankfeifan — APPROVED** [review]
Thanks!

**Review by herve-gross — APPROVED** [review]

File: src/docs/sphinx/advancedExamples/validationStudies/thermoPoromechanics/1DCooling/AnalyticalSol.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        else: 
+            sigma_y = self.C / (1 + self.phi/3.0)
+            epsMech_x = (sigma_y - (3*self.lambdaCoeff + 2*self.shearMod)*epsMech_y )/ (6*self.lambdaCoeff + 4*self.shearMod)
+            multipler = -(sigma_y - (self.lambdaCoeff + 2*self.shearMod)*epsMech_y - 2*self.lambdaCoeff*epsMech_x)/ (4/3 * self.shearMod) 
+            eps_x = epsMech_x + self.thermalExpansionCoeff * deltaTemp
+
+            return sigma_x, sigma_y, eps_x
+        
+    def compute_disp(self, deltaTemp):
+        _, _, esp_x = self.compute_stress(deltaTemp)
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/thermoPoromechanics/1DCooling/AnalyticalSol.py]:
```suggestion
        _, _, eps_x = self.compute_stress(deltaTemp)
```

File: src/docs/sphinx/advancedExamples/validationStudies/thermoPoromechanics/1DCooling/AnalyticalSol.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+        disp_y = 0
+        disp_x = esp_x*1
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/thermoPoromechanics/1DCooling/AnalyticalSol.py]:
```suggestion
        disp_x = eps_x*1
```

**Review by jhuang2601 — APPROVED** [review]
